homeassistant/components/rainforest_eagle/sensor.py

```diff
@@ -0,0 +1,135 @@
+from datetime import timedelta
+import logging
+
+from requests.exceptions import (
+    ConnectionError as ConnectError, HTTPError, Timeout)
+import voluptuous as vol
+
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+from homeassistant.const import (
+    CONF_IP_ADDRESS, CONF_SCAN_INTERVAL, ENERGY_KILO_WATT_HOUR)
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.entity import Entity
+from homeassistant.util import Throttle
+
+CONF_CLOUD_ID = 'cloud_id'
+CONF_INSTALL_CODE = 'install_code'
+POWER_KILO_WATT = 'kW'
+
+_LOGGER = logging.getLogger(__name__)
+
+SCAN_INTERVAL = timedelta(seconds=30)
+
+SENSORS = {
+    "instantanous_demand": (
+        "Eagle-200 Meter Power Demand", POWER_KILO_WATT),
+    "summation_delivered": (
+        "Eagle-200 Total Meter Energy Delivered",
+        ENERGY_KILO_WATT_HOUR),
+    "summation_received": (
+        "Eagle-200 Total Meter Energy Received",
+        ENERGY_KILO_WATT_HOUR),
+    "summation_total": (
+        "Eagle-200 Net Meter Energy (Delivered minus Received)",
+        ENERGY_KILO_WATT_HOUR)
+        }
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_IP_ADDRESS): cv.string,
+    vol.Required(CONF_CLOUD_ID): cv.string,
+    vol.Required(CONF_INSTALL_CODE): cv.string
+})
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Create the Eagle-200 sensor."""
+    ip_address = config[CONF_IP_ADDRESS]
+    cloud_id = config[CONF_CLOUD_ID]
+    install_code = config[CONF_INSTALL_CODE]
+    interval = config.get(CONF_SCAN_INTERVAL, SCAN_INTERVAL)
+
+    eagle_data = EagleData(ip_address, cloud_id, install_code, interval)
+    eagle_data.update()
+
+    monitored_conditions = list(SENSORS)
+    sensors = []
+    for condition in monitored_conditions:
+        sensors.append(EagleSensor(
+            eagle_data, condition, SENSORS[condition][0],
+            SENSORS[condition][1]))
+
+    add_devices(sensors)
+
+
+class EagleSensor(Entity):
+    """Implementation of the Rainforest Eagle-200 sensor."""
+
+    def __init__(
+            self, eagle_data, sensor_type, name, unit):
+        """Initialize the sensor."""
+        self.eagle_data = eagle_data
+        self._type = sensor_type
+        self._name = name
+        self._unit_of_measurement = unit
+        self._state = None
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        return self._name
+
+    @property
+    def state(self):
+        """Return the state of the sensor."""
+        return self._state
+
+    @property
+    def unit_of_measurement(self):
+        """Return the unit of measurement."""
+        return self._unit_of_measurement
+
+    def update(self):
+        """Get the energy information from the Rainforest Eagle."""
+        self.eagle_data.update()
+        data = self.eagle_data.data
+        self._state = self.get_state(data)
+
+    def get_state(self, data):
```

Comment:
that is somethings that should go into Data object and on update:
```
...
self.eagle_data.update()
self._state = self.get_state(data)
```

Reply:
Moved update

Reply:
This still needs to move, as suggested above.

Reply:
move this function into data object and use the code on my first comment inside your sensor update function

Reply:
Moved `get_status` from the Sensor class to the Data class.  After moving it because I'm not using any references to the Data object (`self`) inside the `get_status` method, pylint complained.  So I removed `self` as the first argument, and added the `@staticmethod` decorator:

```
@staticmethod
def get_state(data, sensor_type): 
...
```

Also I changed the call in the Sensor update method to not use the instantiated Data object, since `get_status` is now a static method:
```
...
data = self.eagle_data.data
self._state = EagleData.get_state(data, self._type)
```